src/Mod/Sketcher/Gui/TaskDlgEditSketch.cpp

```diff
@@ -80,6 +89,18 @@ TaskDlgEditSketch::~TaskDlgEditSketch()
     if (it == Content.end()) {
         Content.push_back(SolverAdvanced);
     }
+
+    connectionToolSettings.disconnect();
+}
+
+void TaskDlgEditSketch::slotToolChanged(const std::string& toolname)
+{
+    bool hidden = toolname == "DSH_None" || toolname == "DSH_Point";

```

Comment:
Special behaviour for the widget should be dictated by the tool, rather than hard-coded here.

This a kind of anti-pattern and we should always strive to avoid it.

Reply:
Got it.